internal/backends/postgresql/backend.go

```diff
@@ -28,17 +28,19 @@ import (
 
 // backend implements backends.Backend interface.
 type backend struct {
-	r *metadata.Registry
+	r          *metadata.Registry
+	noContract bool
 }
 
 // NewBackendParams represents the parameters of NewBackend function.
 //
 //nolint:vet // for readability
 type NewBackendParams struct {
-	URI string
-	L   *zap.Logger
-	P   *state.Provider
-	_   struct{} // prevent unkeyed literals
+	URI        string
+	L          *zap.Logger
+	P          *state.Provider
+	NoContract bool
```

Comment:
I'm not sure about this approach, so if you have a better idea please let me know.